nicegui/testing/user_download.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import asyncio
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, List, Optional, Union
+
+import httpx
+
+from .. import background_tasks
+
+if TYPE_CHECKING:
+    from .user import User
+
+
+class UserDownload:
+
+    def __init__(self, user: User) -> None:
+        self.http_responses: List[httpx.Response] = []
+        self.user = user
+
+    def __call__(self, src: Union[str, Path, bytes], filename: Optional[str] = None, media_type: str = '') -> Any:
+        background_tasks.create(self._get(src))
+
+    async def _get(self,  src: Union[str, Path, bytes]) -> None:
+        if isinstance(src, bytes):
+            await asyncio.sleep(0)
+            response = httpx.Response(200, content=src)
+        else:
+            response = await self.user.http_client.get(str(src))
+        self.http_responses.append(response)
+
+    async def new(self) -> httpx.Response:
```

Comment:
I wonder - what's the reasoning behind this returning a `httpx.Response`, as opposed to some other object that just encapsulates e.g. the filename and contents?

Reply:
A custom encapsulation would require documentation and maintenance. And because the `Response` object has a fairly nice API and allows to even check for response headers etc. it is good suited for tests.